### ISSUE TITLE
Introduce events

### ISSUE THREAD

**Issue by Pierstoval** [usefathom/fathom#61]
For now, the tracking system is only associated with page view.

Is there any way to introduce a `fathom.event('eventName', data);` function in the JS lib that would create events in the database?

Sorry for posting this without having looked at the code, but I don't speak golang 😜

**Comment by dannyvankooten** [comment]
Hey @Pierstoval, thanks for chiming in!

Events is definitely on the list, most likely even next on the list. We think that after pageviews and referrers, events is one of the more important metrics for a lot of us website owners.

We'll be setting up a public roadmap soon where everyone can follow along on our progress and vote for features etc. We'll be using that to guide us in what should be included for our next version.

Cheers!

**Comment by nektro** [comment]
I haven't personally used events much in analytics, but I have 2 questions about this topic.
1) would implementing this go out of scope of fathom being data-light
2) could this be used to implement #50

**Comment by bradrydzewski** [comment]
@dannyvankooten if you are looking for some help with this feature I would be interested. Do you have any specific design in mind that you can share?

**Comment by bitboxer** [comment]
Would also love to help here. This is a key feature that i need to have before switching one of my side projects to fathom. With events and some stats i can do a/b testing and other things.

**Comment by avorio** [comment]
+1

Count on me for testing and contributing with use cases and feedback.

**Comment by Pierstoval** [comment]
The advantage is that an event should be "as simple as" an object with:

* Datetime
* User id if specified
* Event name
* Optional data (maybe a JSON or something)

Adding such event shouldn't be that hard, it's just that I don't write Go and therefore don't know how to contribute 😅

**Comment by joshmarinacci** [comment]
Any updates on events? It would be super useful.

**Comment by m1guelpf** [comment]
Hey folks!

In case you aren't aware, this repository isn't currently mantained. I think the Fathom team has plans to continue the support at some point in the future, but can't say for certain. If events are a hard requirement for you, you should probably check [Fathom v2](https://usefathom.com) out, the paid version of this project, which does include support for events. You can also try to make a PR to this repo, as it's far easier to review code than to draft and write and it'll probably increase the changes of this getting in 😄

**Comment by nektro** [comment]
Darn, didn't even last 2 years.. :/

**Comment by alexanderadam** [comment]
> Darn, didn't even last 2 years.. :/

There are alternatives [like plausible](https://github.com/plausible-insights/plausible) out there.
Open source often finds a way…

**Comment by nektro** [comment]
>  Open source often finds a way…

fair enough, was just sad to see since I've been watching this repo for a long time. I'll prolly end up making my own tbh, but that's off-topic for these comments.

**Comment by Pierstoval** [comment]
I'm also sad of it, I really liked the idea of being able to self-host a simple way to make anonymous analytics (that isn't yet blocked by ad-blockers).

**Comment by JackEllis** [comment]
Closed as Miguel already answered the question perfectly. Thanks Miguel.
